packages/Chem/src/chem-searches.ts

```diff
@@ -84,116 +90,104 @@ function _chemGetDiversities(limit: number, molStringsColumn: DG.Column, fingerp
   return diversities;
 }
 
+
 function invalidatedColumnKey(col: DG.Column): string {
   return col.dataFrame?.name + '.' + col.name;
 }
 
-function colInvalidated(col: DG.Column, createMols: boolean): Boolean {
+function colInvalidated(col: DG.Column): Boolean {
   return (lastColumnInvalidated == invalidatedColumnKey(col) &&
-    col.getTag(FING_COL_TAGS.invalidatedForVersion) == String(col.version) &&
-    (!createMols || col.getTag(FING_COL_TAGS.molsCreatedForVersion) == String(col.version)));
+    col.getTag(FING_COL_TAGS.molsCreatedForVersion) == String(col.version));
 }
 
-async function _invalidate(molCol: DG.Column, createMols: boolean) {
-  if (!colInvalidated(molCol, createMols)) {
-    if (createMols) {
-      await (await getRdKitService()).initMoleculesStructures(molCol.toList());
-      molCol.setTag(FING_COL_TAGS.molsCreatedForVersion, String(molCol.version + 2));
-    }
+//updates array of RDMols in case column has been changed
+async function _invalidate(molCol: DG.Column) {
+  if (!colInvalidated(molCol)) {
+    await (await getRdKitService()).initMoleculesStructures(molCol.toList());
+    molCol.setTag(FING_COL_TAGS.molsCreatedForVersion, String(molCol.version + 1));
     lastColumnInvalidated = invalidatedColumnKey(molCol);
-    molCol.setTag(FING_COL_TAGS.invalidatedForVersion, String(molCol.version + 1));
   }
 }
 
-function checkForFingerprintsColumn(col: DG.Column, fingerprintsType: Fingerprint): Uint8Array[] | null {
-  const colNameTag = '.' + fingerprintsType + '.Column';
-  const colVerTag = '.' + fingerprintsType + '.Version';
 
-  if (col.getTag(colNameTag) &&
-      col.getTag(colVerTag) == String(col.version) &&
-      col.dataFrame) {
-    const fingCol = col.dataFrame.columns.byName(col.getTag(colNameTag));
-    if (fingCol)
-      return fingCol.toList();
+function checkForSavedColumn(col: DG.Column, colTag: Fingerprint | string): DG.Column<any> | null{
+  const savedColName = '~' + col.name + '.' + colTag;
+  const colVerTag = '.' + colTag + '.Version';
+
+  if (col.dataFrame && col.dataFrame?.col(savedColName) &&
+      col.getTag(colVerTag) == String(col.version)) {
+    const savedCol = col.dataFrame.columns.byName(savedColName);
+    return savedCol;
   }
   return null;
 }
 
-function saveFingerprintsToCol(col: DG.Column, fgs: Uint8Array[],
-  fingerprintsType: Fingerprint, createdMols: boolean): void {
+function saveColumns(col: DG.Column, data: ((Uint8Array | null)[] | (string | null)[])[],
```

Comment:
This needs to be documented. And this smells bad to me: `data: ((Uint8Array | null)[] | (string | null)[])[]`

Reply:
There can be 2 types of columns we want to save - fp or canonical smiles. Should I make this a generic function instead of indicating the exact types?